### DISCUSSION TITLE
Customize Entire or the Root of Web URL

### DISCUSSION THREAD

**Discussion by itswzyss** [iterate-ch/cyberduck#16487] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I use CloudFlare R2 to host data. The domain I'm working with is `data.wzyss.dev`. The `data` R2 bucket is directly attached to the domain as a CNAME. Unfortunately, CyberDuck includes the bucket name in the URL -- which is a problem for me.

If I use `data.wzyss.dev` as the custom Web URL and I upload an item named `test.txt` to the bucket, the URL should be `data.wzyss.dev/test.txt`. However, what I get from CyberDuck is `data.wzyss.dev/data/test.txt`, which won't work.

This requires me to remove the extra `/data/` from the URL before it can be used.

**Describe the solution you'd like**
If you could customize where the "root" of the directory would be, that might solve this problem. I'm not sure what else I can do (other than completely modify my domain setup) to achieve this.

**Describe alternatives you've considered**
The only workaround / alternative I can think of right now is to manually edit the URL each time.

**Comment by AliveDevil** [comment, marked as answer]
This works just fine for me, when setting a default path.

**Reply by dkocher** [reply]
Follow the [documentation](https://docs.cyberduck.io/cyberduck/bookmarks/#web-url). In your example, configure the bookmark with the bucket name in _Path_.

* _Path_: `/data`
* _Web URL: `https://data.wzyss.dev/`

**Reply by itswzyss** [reply]
Thanks for this... Somehow this just didn't occur to me. I've been using master credentials that display all buckets in a single bookmark. I see I can fix this by using a bookmark for each bucket, and setting the path and web URL respectively.

Thanks so much! I've been doing this the hard way for over a year...
